@here/harp-mapview/lib/MapView.ts

```diff
@@ -3503,15 +3503,16 @@ export class MapView extends THREE.EventDispatcher {
         // running. The initial placement of text in this render call may have changed the loading
         // state of the TextElementsRenderer, so this has to be checked again.
         if (
-            !this.m_firstFrameComplete &&
+            !this.textElementsRenderer.loading &&
+            this.m_visibleTiles.allVisibleTilesLoaded &&
             this.m_initialTextPlacementDone &&
-            !this.isDynamicFrame &&
-            !this.textElementsRenderer.loading
+            !this.m_animatedExtrusionHandler.isAnimating
         ) {
-            this.m_firstFrameComplete = true;
-
-            if (gatherStatistics) {
-                stats.appResults.set("firstFrameComplete", frameStartTime);
+            if (this.m_firstFrameComplete) {
+                this.m_firstFrameComplete = true;
```

Comment:
Not needed.

Reply:
This is still missing, i.e. if m_firstFrameComplete is true, we don't need to set it true

Reply:
Oh, should be `if (this.m_firstFrameComplete === false)` instead. Fixed it